From 1505e0f162d3771406d2a741e3ea245af99accf2 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Thu, 12 Sep 2013 18:50:49 +1000
Subject: [PATCH] add forum link to device info

Change-Id: I94d372a1d2c45fef4e543bef021f073dfe4deedb

diff --git a/res/values/cm_strings.xml b/res/values/cm_strings.xml
index 6eacfdc..b017ee4 100644
--- a/res/values/cm_strings.xml
+++ b/res/values/cm_strings.xml
@@ -14,6 +14,11 @@
      limitations under the License.
 -->
 <resources xmlns:xliff="urn:oasis:names:tc:xliff:document:1.2">
+    <!-- Konsta forum link -->
+    <string name="konsta_web_title">CyanogenMod10.2-PenguinKANG</string>
+    <string name="konsta_web_summary">Visit MoDaCo thread</string>
+    <string name="konsta_web_url">http://www.modaco.com/topic/365005-devrom-cyanogenmod-102-android-43-for-zte-v9/</string>
+
     <!-- Activity label of BluetoothMasPermissionActivity, also used as Strings in the permission dialog [CHAR LIMIT=none] -->
     <string name="bluetooth_mas_request">"Message Access request"</string>
     <!-- Bluetooth MAS permission Alert Activity text [CHAR LIMIT=none] -->
diff --git a/res/xml/device_info_settings.xml b/res/xml/device_info_settings.xml
index 2a800a0..c71951c 100644
--- a/res/xml/device_info_settings.xml
+++ b/res/xml/device_info_settings.xml
@@ -161,4 +161,9 @@
                 android:title="@string/selinux_status"
                 android:summary="@string/selinux_status_enforcing"/>
 
+        <!-- Konsta forum link -->
+        <Preference android:key="konsta_web"
+                android:title="@string/konsta_web_title"
+                android:summary="@string/konsta_web_summary"/>
+
 </PreferenceScreen>
diff --git a/src/com/android/settings/DeviceInfoSettings.java b/src/com/android/settings/DeviceInfoSettings.java
index c86a724..e068635 100644
--- a/src/com/android/settings/DeviceInfoSettings.java
+++ b/src/com/android/settings/DeviceInfoSettings.java
@@ -21,6 +21,7 @@ import android.app.AlertDialog;
 import android.content.Context;
 import android.content.DialogInterface;
 import android.content.Intent;
+import android.net.Uri;
 import android.os.Build;
 import android.os.Bundle;
 import android.os.SELinux;
@@ -30,6 +31,7 @@ import android.os.UserHandle;
 import android.preference.Preference;
 import android.preference.PreferenceGroup;
 import android.preference.PreferenceScreen;
+import android.preference.Preference.OnPreferenceClickListener;
 import android.util.Log;
 import android.widget.Toast;
 
@@ -159,6 +161,16 @@ public class DeviceInfoSettings extends SettingsPreferenceFragment {
         // Remove regulatory information if not enabled.
         removePreferenceIfBoolFalse(KEY_REGULATORY_INFO,
                 R.bool.config_show_regulatory_info);
+
+        findPreference("konsta_web").setOnPreferenceClickListener(
+            new OnPreferenceClickListener() {
+                public boolean onPreferenceClick(Preference preference) {
+                    Intent browserIntent = new Intent("android.intent.action.VIEW",
+                            Uri.parse(getString(R.string.konsta_web_url)));
+                    startActivity(browserIntent);
+                    return true;
+                }
+            });
     }
 
     @Override
