From 33dc63664a92530113c1b2d6a8c0e8199c9a46e9 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Sun, 5 Jan 2014 16:20:14 +1100
Subject: [PATCH] exclude metadata from ics compat

Change-Id: Ic1a3e7dba98864646b13b5883a1ffc5a644b9488

diff --git a/core/jni/android_hardware_Camera.cpp b/core/jni/android_hardware_Camera.cpp
index 17967e0..12b0d2f 100644
--- a/core/jni/android_hardware_Camera.cpp
+++ b/core/jni/android_hardware_Camera.cpp
@@ -710,6 +710,7 @@ static void android_hardware_Camera_setHasPreviewCallback(JNIEnv *env, jobject t
 static void android_hardware_Camera_setMetadataCb(JNIEnv *env, jobject thiz, jboolean mode)
 {
     ALOGV("setMetadataCb: mode:%d", (int)mode);
+#if defined(QCOM_HARDWARE) && !defined(ICS_CAMERA_BLOB)
     JNICameraContext* context;
     status_t rc;
     sp<Camera> camera = get_native_camera(env, thiz, &context);
@@ -723,6 +724,7 @@ static void android_hardware_Camera_setMetadataCb(JNIEnv *env, jobject thiz, jbo
     if (rc != NO_ERROR) {
         jniThrowException(env, "java/lang/RuntimeException", "set metadata mode failed");
     }
+#endif
 }
 
 static void android_hardware_Camera_addCallbackBuffer(JNIEnv *env, jobject thiz, jbyteArray bytes, int msgType) {
