From a38661042005fb69719016c858f3fb5430cc3dca Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Mon, 3 Mar 2014 16:31:23 +1100
Subject: [PATCH] Revert "audioflinger: Set power hints for audio activity"

This reverts commit b4be352d74248987c74cd75b49c82c2664b89a60.

diff --git a/services/audioflinger/AudioFlinger.h b/services/audioflinger/AudioFlinger.h
index 3407a4c..b861e4f 100644
--- a/services/audioflinger/AudioFlinger.h
+++ b/services/audioflinger/AudioFlinger.h
@@ -51,7 +51,6 @@
 #include <system/audio.h>
 #include <hardware/audio.h>
 #include <hardware/audio_policy.h>
-#include <hardware/power.h>
 
 #include <media/AudioBufferProvider.h>
 #include <media/ExtendedAudioBufferProvider.h>
diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 5e52a4b..5b21708 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -63,7 +63,6 @@
 #include <media/nbaio/SourceAudioBufferProvider.h>
 
 #include <powermanager/PowerManager.h>
-#include <hardware/power.h>
 
 #include <common_time/cc_helper.h>
 #include <common_time/local_clock.h>
@@ -513,14 +512,6 @@ void AudioFlinger::ThreadBase::dumpEffectChains(int fd, const Vector<String16>&
     }
 }
 
-void AudioFlinger::ThreadBase::setPowerHint(bool active)
-{
-    if (mPowerModule && mPowerModule->powerHint) {
-        mPowerModule->powerHint(mPowerModule, POWER_HINT_AUDIO,
-                active ? (void *)"state=1" : (void *)"state=0");
-    }
-}
-
 void AudioFlinger::ThreadBase::acquireWakeLock(int uid)
 {
     Mutex::Autolock _l(mLock);
@@ -550,7 +541,6 @@ void AudioFlinger::ThreadBase::acquireWakeLock_l(int uid)
 {
     getPowerManager_l();
     if (mPowerManager != 0) {
-        setPowerHint(true);
         sp<IBinder> binder = new BBinder();
         status_t status;
         if (uid >= 0) {
@@ -587,7 +577,6 @@ void AudioFlinger::ThreadBase::releaseWakeLock_l()
         }
         mWakeLockToken.clear();
     }
-    setPowerHint(false);
 }
 
 void AudioFlinger::ThreadBase::updateWakeLockUids(const SortedVector<int> &uids) {
@@ -608,10 +597,6 @@ void AudioFlinger::ThreadBase::getPowerManager_l() {
             binder->linkToDeath(mDeathRecipient);
         }
     }
-
-    if (mPowerModule == 0) {
-        hw_get_module(POWER_HARDWARE_MODULE_ID, (const hw_module_t **)&mPowerModule);
-    }
 }
 
 void AudioFlinger::ThreadBase::updateWakeLockUids_l(const SortedVector<int> &uids) {
@@ -632,7 +617,6 @@ void AudioFlinger::ThreadBase::updateWakeLockUids_l(const SortedVector<int> &uid
 void AudioFlinger::ThreadBase::clearPowerManager()
 {
     Mutex::Autolock _l(mLock);
-    setPowerHint(false);
     releaseWakeLock_l();
     mPowerManager.clear();
 }
diff --git a/services/audioflinger/Threads.h b/services/audioflinger/Threads.h
index 77fb09b..8ebfa00 100644
--- a/services/audioflinger/Threads.h
+++ b/services/audioflinger/Threads.h
@@ -247,7 +247,6 @@ protected:
                 void        releaseWakeLock_l();
                 void        updateWakeLockUids(const SortedVector<int> &uids);
                 void        updateWakeLockUids_l(const SortedVector<int> &uids);
-                void        setPowerHint(bool active);
                 void        getPowerManager_l();
                 void setEffectSuspended_l(const effect_uuid_t *type,
                                           bool suspend,
@@ -332,7 +331,6 @@ protected:
                                         mSuspendedSessions;
                 static const size_t     kLogSize = 4 * 1024;
                 sp<NBLog::Writer>       mNBLogWriter;
-                power_module_t          *mPowerModule;
 };
 
 // --- PlaybackThread ---
