From f88181c5641eff3c1840a03453d86fdc3eeee116 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Sun, 9 Mar 2014 15:15:31 +1100
Subject: [PATCH] fix build error with no ion

Change-Id: Ie4aaef6374eb8dfe9557ca6c3b1ef00947028809

diff --git a/media/libstagefright/ExtendedUtils.cpp b/media/libstagefright/ExtendedUtils.cpp
index fe04f91..f18a7e2 100644
--- a/media/libstagefright/ExtendedUtils.cpp
+++ b/media/libstagefright/ExtendedUtils.cpp
@@ -54,10 +54,12 @@ static const int64_t kMaxAVSyncLateMargin     = 250000;
 
 #include "include/ExtendedExtractor.h"
 #include "include/avc_utils.h"
+#if defined(ION_IOC_DRAIN) || defined(ION_IOC_PREFETCH)
 #include <fcntl.h>
 #include <linux/msm_ion.h>
 #define MEM_DEVICE "/dev/ion"
 #define MEM_HEAP_ID ION_CP_MM_HEAP_ID
+#endif
 
 namespace android {
 
