From 62fd4df53403f16821eabb06f8e44adfaa2f2000 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Mon, 2 Dec 2013 12:09:11 +1100
Subject: [PATCH] exclude metadata etc. from ics compat

Change-Id: Ib230daa73fc868fc646f5dfb44aa45123f3ff888

diff --git a/services/camera/libcameraservice/api1/CameraClient.cpp b/services/camera/libcameraservice/api1/CameraClient.cpp
index d63eb3c..7e9a7a8 100644
--- a/services/camera/libcameraservice/api1/CameraClient.cpp
+++ b/services/camera/libcameraservice/api1/CameraClient.cpp
@@ -701,6 +701,7 @@ status_t CameraClient::sendCommand(int32_t cmd, int32_t arg1, int32_t arg2) {
         enableMsgType(CAMERA_MSG_STATS_DATA);
     } else if (cmd == CAMERA_CMD_HISTOGRAM_OFF) {
         disableMsgType(CAMERA_MSG_STATS_DATA);
+#ifndef QCOM_ICS_COMPAT
     } else if (cmd == CAMERA_CMD_METADATA_ON) {
         enableMsgType(CAMERA_MSG_META_DATA);
     } else if (cmd == CAMERA_CMD_METADATA_OFF) {
@@ -712,6 +713,7 @@ status_t CameraClient::sendCommand(int32_t cmd, int32_t arg1, int32_t arg2) {
         disableMsgType(CAMERA_MSG_SHUTTER);
         disableMsgType(CAMERA_MSG_COMPRESSED_IMAGE);
 #endif
+#endif
     }
 
     return mHardware->sendCommand(cmd, arg1, arg2);
