diff --git a/driver/rsdCore.cpp b/driver/rsdCore.cpp
index b578d41..f325087 100644
--- a/driver/rsdCore.cpp
+++ b/driver/rsdCore.cpp
@@ -291,10 +291,9 @@ void Shutdown(Context *rsc) {
     for (uint32_t ct = 0; ct < dc->mWorkers.mCount; ct++) {
         dc->mWorkers.mLaunchSignals[ct].set();
     }
-    int status;
     void *res;
     for (uint32_t ct = 0; ct < dc->mWorkers.mCount; ct++) {
-        status = pthread_join(dc->mWorkers.mThreadId[ct], &res);
+        pthread_join(dc->mWorkers.mThreadId[ct], &res);
     }
     rsAssert(android_atomic_acquire_load(&dc->mWorkers.mRunningCount) == 0);
 
