From cf21706148c84a0886e1ef88140ae167da746940 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Tue, 29 Oct 2013 18:34:22 +1100
Subject: [PATCH] fix renderscript on armv6

Change-Id: Iebd19858892ba9c9a4c735292527bdb4fc34f961

diff --git a/include/bcc/Config/Config.h b/include/bcc/Config/Config.h
index 0331ae0..230f464 100644
--- a/include/bcc/Config/Config.h
+++ b/include/bcc/Config/Config.h
@@ -56,8 +56,13 @@
   #endif
 #endif
 
-#define DEFAULT_ARM_TRIPLE_STRING      "armv7-none-linux-gnueabi"
-#define DEFAULT_THUMB_TRIPLE_STRING    "thumbv7-none-linux-gnueabi"
+#if defined(TARGET_CPU_VARIANT_ARM11)
+  #define DEFAULT_ARM_TRIPLE_STRING      "armv6-none-linux-gnueabi"
+  #define DEFAULT_THUMB_TRIPLE_STRING    "thumbv6-none-linux-gnueabi"
+#else
+  #define DEFAULT_ARM_TRIPLE_STRING      "armv7-none-linux-gnueabi"
+  #define DEFAULT_THUMB_TRIPLE_STRING    "thumbv7-none-linux-gnueabi"
+#endif
 #define DEFAULT_MIPS_TRIPLE_STRING     "mipsel-none-linux-gnueabi"
 #define DEFAULT_X86_TRIPLE_STRING      "i686-unknown-linux"
 #define DEFAULT_X86_64_TRIPLE_STRING   "x86_64-unknown-linux"
diff --git a/lib/Renderscript/runtime/build_bc_lib.mk b/lib/Renderscript/runtime/build_bc_lib.mk
index 4e2f1aa..c5368c7 100644
--- a/lib/Renderscript/runtime/build_bc_lib.mk
+++ b/lib/Renderscript/runtime/build_bc_lib.mk
@@ -30,9 +30,15 @@ bc_cflags := -MD \
              -c \
              -O3 \
              -fno-builtin \
-             -emit-llvm \
-             -target armv7-none-linux-gnueabi \
-             -fsigned-char \
+             -emit-llvm
+
+ifeq ($(TARGET_CPU_VARIANT),arm11)
+bc_cflags += -target armv6-none-linux-gnueabi
+else
+bc_cflags += -target armv7-none-linux-gnueabi
+endif
+
+bc_cflags += -fsigned-char \
              $(bc_translated_clang_cc1_cflags)
 
 ifeq ($(rs_debug_runtime),1)
diff --git a/lib/Support/TargetCompilerConfigs.cpp b/lib/Support/TargetCompilerConfigs.cpp
index 7d3de45..6d51849 100644
--- a/lib/Support/TargetCompilerConfigs.cpp
+++ b/lib/Support/TargetCompilerConfigs.cpp
@@ -46,9 +46,14 @@ void
 ARMBaseCompilerConfig::GetFeatureVector(std::vector<std::string> &pAttributes,
                                         bool pInThumbMode, bool pEnableNEON) {
 #if defined(ARCH_ARM_HAVE_VFP)
+#  if defined(TARGET_CPU_VARIANT_ARM11)
+  pAttributes.push_back("+vfp2");
+  pAttributes.push_back("-d16");
+#  else
   pAttributes.push_back("+vfp3");
-#  if !defined(ARCH_ARM_HAVE_VFP_D32)
+#    if !defined(ARCH_ARM_HAVE_VFP_D32)
   pAttributes.push_back("+d16");
+#    endif
 #  endif
 #endif
 
diff --git a/libbcc-device-build.mk b/libbcc-device-build.mk
index aead9c1..20b1952 100644
--- a/libbcc-device-build.mk
+++ b/libbcc-device-build.mk
@@ -42,6 +42,9 @@ ifeq ($(TARGET_ARCH),arm)
       LOCAL_CFLAGS += -DARCH_ARM_HAVE_VFP_D32
     endif
   endif
+  ifeq ($(TARGET_CPU_VARIANT),arm11)
+    LOCAL_CFLAGS += -DTARGET_CPU_VARIANT_ARM11
+  endif
   ifeq ($(ARCH_ARM_HAVE_NEON),true)
     # Disable NEON on cortex-a15 temporarily
     ifneq ($(strip $(TARGET_CPU_VARIANT)), cortex-a15)
diff --git a/tools/bcc_compat/Android.mk b/tools/bcc_compat/Android.mk
index b8be047..5a38e5b 100644
--- a/tools/bcc_compat/Android.mk
+++ b/tools/bcc_compat/Android.mk
@@ -37,6 +37,10 @@ LOCAL_C_INCLUDES := \
 LOCAL_LDLIBS = -ldl
 LOCAL_SRC_FILES := Main.cpp
 
+ifeq ($(TARGET_CPU_VARIANT),arm11)
+  LOCAL_CFLAGS += -DTARGET_CPU_VARIANT_ARM11
+endif
+
 include $(LIBBCC_HOST_BUILD_MK)
 include $(LIBBCC_GEN_CONFIG_MK)
 include $(LLVM_HOST_BUILD_MK)
diff --git a/tools/bcc_compat/Main.cpp b/tools/bcc_compat/Main.cpp
index d0a6f27..b074f64 100644
--- a/tools/bcc_compat/Main.cpp
+++ b/tools/bcc_compat/Main.cpp
@@ -179,8 +179,13 @@ bool ConfigCompiler(RSCompilerDriver &pCompilerDriver) {
   // Explicitly set ARM feature vector
   if (config->getTriple().find("arm") != std::string::npos) {
     std::vector<std::string> fv;
+#if defined(TARGET_CPU_VARIANT_ARM11)
+    fv.push_back("+vfp2");
+    fv.push_back("-d16");
+#else
     fv.push_back("+vfp3");
     fv.push_back("+d16");
+#endif
     fv.push_back("-neon");
     fv.push_back("-neonfp");
     config->setFeatureString(fv);
