From 5fef138d9dbfec1aaecafee9d7bef0f2fb9e7a83 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Mon, 14 Oct 2013 16:08:22 +1100
Subject: [PATCH] fix early bootanimation distortion on 90/270 displays

Change-Id: Icfde9a514644c24086d3ce2417b6e5438e94cec2

diff --git a/services/surfaceflinger/DisplayDevice.cpp b/services/surfaceflinger/DisplayDevice.cpp
index fce66c8..4c2832d 100644
--- a/services/surfaceflinger/DisplayDevice.cpp
+++ b/services/surfaceflinger/DisplayDevice.cpp
@@ -393,6 +393,9 @@ void DisplayDevice::setProjection(int orientation,
         // the destination frame can be invalid if it has never been set,
         // in that case we assume the whole display frame.
         frame = Rect(w, h);
+        if (R.getOrientation() & Transform::ROT_90) {
+            swap(frame.right, frame.bottom);
+        }
     }
 
     if (viewport.isEmpty()) {
