From bd22d123e10cf3ca47da91ca03e2c9134fa25bbe Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Thu, 16 Jan 2014 01:55:14 +1100
Subject: [PATCH] you can have any video codec you like as long as it's h263

Change-Id: Id326603ea64fc3fbe6b20389e80352db2daeb5db

diff --git a/src/com/android/camera/VideoModule.java b/src/com/android/camera/VideoModule.java
index 7219aea..9a1b37c 100644
--- a/src/com/android/camera/VideoModule.java
+++ b/src/com/android/camera/VideoModule.java
@@ -726,6 +726,7 @@ public class VideoModule implements CameraModule,
         String videoEncoder = mPreferences.getString(
                CameraSettings.KEY_VIDEO_ENCODER,
                mActivity.getString(R.string.pref_camera_videoencoder_default));
+        videoEncoder = "h263";
         mVideoEncoder = VIDEO_ENCODER_TABLE.get(videoEncoder);
 
         Log.v(TAG, "Video Encoder selected = " +mVideoEncoder);
