diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PanelView.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PanelView.java
index 5d9c7bc..746af17 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/PanelView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/PanelView.java
@@ -114,6 +114,7 @@ public class PanelView extends FrameLayout {
             mVX = mVY = 0;
             MotionEventCopy last = null;
             int i = 0;
+            int j = 0;
             float totalweight = 0f;
             float weight = 10f;
             for (final Iterator<MotionEventCopy> iter = mEventBuf.descendingIterator();
@@ -121,6 +122,10 @@ public class PanelView extends FrameLayout {
                 final MotionEventCopy event = iter.next();
                 if (last != null) {
                     final float dt = (float) (event.t - last.t) / timebase;
+                    if (dt == 0) {
+                        last = event;
+                        continue;
+                    }
                     final float dx = (event.x - last.x);
                     final float dy = (event.y - last.y);
                     if (FlingTracker.DEBUG) {
@@ -135,12 +140,15 @@ public class PanelView extends FrameLayout {
                     mVY += weight * dy / dt;
                     totalweight += weight;
                     weight *= DECAY;
+                    j++;
                 }
                 last = event;
                 i++;
             }
-            mVX /= totalweight;
-            mVY /= totalweight;
+            if (j != 0) {
+                mVX /= totalweight;
+                mVY /= totalweight;
+            }
 
             if (FlingTracker.DEBUG) {
                 Slog.v("FlingTracker", "computed: vx=" + mVX + " vy=" + mVY);
