From a7de02ed2fd361bd44fb1d30da290fccbac005d3 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Tue, 29 Oct 2013 18:43:23 +1100
Subject: [PATCH] fix renderscript on armv6

Change-Id: I785a97ff8976896f99ff4725a7ccba133876c3d8

diff --git a/core/llvm_config.mk b/core/llvm_config.mk
index c4b4af4..6e1b8bb 100644
--- a/core/llvm_config.mk
+++ b/core/llvm_config.mk
@@ -20,7 +20,11 @@ CLANG_CONFIG_UNKNOWN_CFLAGS := \
   -funswitch-loops
 
 ifeq ($(TARGET_ARCH),arm)
-  RS_TRIPLE := armv7-none-linux-gnueabi
+  ifeq ($(TARGET_CPU_VARIANT),arm11)
+    RS_TRIPLE := armv6-none-linux-gnueabi
+  else
+    RS_TRIPLE := armv7-none-linux-gnueabi
+  endif
   CLANG_CONFIG_EXTRA_ASFLAGS += \
     -target arm-linux-androideabi \
     -nostdlibinc \
