From 1d2385577d89ebeef57dc72489f185e720adfa85 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Wed, 3 Apr 2013 21:26:39 +0300
Subject: [PATCH] don't copy sample videos

Change-Id: I9140d38a521601a1e9fc044b88de94308ae1687f

diff --git a/config/common_full.mk b/config/common_full.mk
index ad73472..1919b52 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -7,9 +7,6 @@ include frameworks/base/data/sounds/NewAudio.mk
 # Extra Ringtones
 include frameworks/base/data/sounds/AudioPackageNewWave.mk
 
-# Bring in all video files
-$(call inherit-product, frameworks/base/data/videos/VideoPackage2.mk)
-
 # Include CM audio files
 include vendor/cm/config/cm_audio.mk
 
