From ffc79fbe296367e24d7aee1726ca097b0e60b6ab Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Tue, 16 Oct 2012 21:32:22 +0300
Subject: [PATCH] Revert "cm: Build the commandline tools"

This reverts commit 1949eb5c59fdc1c0fcc2cb157d486fa97f972e82.

diff --git a/config/common.mk b/config/common.mk
index 344d6b1..fcb5f1d 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -189,18 +189,7 @@ PRODUCT_PACKAGES += \
     openvpn \
     e2fsck \
     mke2fs \
-    tune2fs \
-    bash \
-    vim \
-    nano \
-    htop \
-    powertop \
-    lsof \
-    mount.exfat \
-    fsck.exfat \
-    mkfs.exfat \
-    ntfsfix \
-    ntfs-3g
+    tune2fs
 
 # Openssh
 PRODUCT_PACKAGES += \
