From 28e1ef274a0df78166faf81adeacb8934e62f0f2 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Mon, 30 Dec 2013 01:37:52 +1100
Subject: [PATCH] remove some unneeded cm specific packages from build

Change-Id: Ia182425f8b86756652b7ace7a12c586f7cd4f6b9

diff --git a/config/common.mk b/config/common.mk
index e3c5061..566dbaa 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -97,11 +97,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/etc/init.local.rc:root/init.cm.rc
 
-# Bring in camera effects
-PRODUCT_COPY_FILES +=  \
-    vendor/cm/prebuilt/common/media/LMprec_508.emd:system/media/LMprec_508.emd \
-    vendor/cm/prebuilt/common/media/PFFprec_600.emd:system/media/PFFprec_600.emd
-
 # Enable SIP+VoIP on all targets
 PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml
@@ -125,7 +120,6 @@ PRODUCT_PACKAGES += \
 
 # Optional CM packages
 PRODUCT_PACKAGES += \
-    VoicePlus \
     VoiceDialer \
     SoundRecorder \
     Basic \
@@ -139,12 +133,9 @@ PRODUCT_PACKAGES += \
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
-    CMWallpapers \
     Apollo \
     CMFileManager \
     LockClock \
-    CMUpdater \
-    CMFota \
     CMAccount
 
 # CM Hardware Abstraction Framework
diff --git a/config/common_full.mk b/config/common_full.mk
index 2476503..d68e710 100644
--- a/config/common_full.mk
+++ b/config/common_full.mk
@@ -3,23 +3,3 @@ $(call inherit-product, vendor/cm/config/common.mk)
 
 # Include CM audio files
 include vendor/cm/config/cm_audio.mk
-
-# Optional CM packages
-PRODUCT_PACKAGES += \
-    Galaxy4 \
-    HoloSpiralWallpaper \
-    LiveWallpapers \
-    LiveWallpapersPicker \
-    MagicSmokeWallpapers \
-    NoiseField \
-    PhaseBeam \
-    VisualizationWallpapers \
-    PhotoTable
-
-PRODUCT_PACKAGES += \
-    VideoEditor \
-    libvideoeditor_jni \
-    libvideoeditor_core \
-    libvideoeditor_osal \
-    libvideoeditor_videofilters \
-    libvideoeditorplayer
