From 34f23fb44beb86459126b09f7b24864ad67295d8 Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Mon, 9 Dec 2013 15:37:10 +1100
Subject: [PATCH] remove some unneeded cm specific packages from build

Change-Id: Iff28f576d8996d4b83efdb4de2b1a5294a0d6d0f

diff --git a/config/common.mk b/config/common.mk
index f2ba68d..9f81289 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -118,11 +118,6 @@ PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/bin/compcache:system/bin/compcache \
     vendor/cm/prebuilt/common/bin/handle_compcache:system/bin/handle_compcache
 
-# Bring in camera effects
-PRODUCT_COPY_FILES +=  \
-    vendor/cm/prebuilt/common/media/LMprec_508.emd:system/media/LMprec_508.emd \
-    vendor/cm/prebuilt/common/media/PFFprec_600.emd:system/media/PFFprec_600.emd
-
 # Enable SIP+VoIP on all targets
 PRODUCT_COPY_FILES += \
     frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml
@@ -161,8 +156,6 @@ PRODUCT_PACKAGES += \
     Apollo \
     CMFileManager \
     LockClock \
-    CMUpdater \
-    CMFota \
     CMAccount \
     WhisperPush
 
diff --git a/config/common_mini_phone.mk b/config/common_mini_phone.mk
index eff08fa..65689ff 100644
--- a/config/common_mini_phone.mk
+++ b/config/common_mini_phone.mk
@@ -4,9 +4,6 @@ $(call inherit-product, vendor/cm/config/common.mk)
 # Bring in all audio files
 include frameworks/base/data/sounds/OldAudio.mk
 
-# Include CM audio files
-include vendor/cm/config/cm_audio.mk
-
 # Default ringtone
 PRODUCT_PROPERTY_OVERRIDES += \
     ro.config.ringtone=Ring_Digital_02.ogg \
