From bad0aceb5ccf630e0386232206fd81b2855bd920 Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Sat, 7 Sep 2013 21:41:04 +1000
Subject: [PATCH] buildfix: get rid of otablock

Change-Id: I2e25a7e6e755a20f9c62ec4e50eba5a032e1f249

diff --git a/config/common.mk b/config/common.mk
index 17a2f68..b01bccc 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -115,10 +115,6 @@ PRODUCT_COPY_FILES += \
 PRODUCT_COPY_FILES += \
     vendor/cm/prebuilt/common/bin/modelid_cfg.sh:system/bin/modelid_cfg.sh
 
-# block stock OTAs
-PRODUCT_COPY_FILES += \
-    vendor/cm/prebuilt/common/bin/otablock:system/bin/otablock
-
 # Terminal Emulator
 PRODUCT_COPY_FILES +=  \
     vendor/cm/proprietary/Term.apk:system/app/Term.apk \
