From ad704eba966be60888d4c49cbc2850103f4ddceb Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Wed, 9 Oct 2013 14:40:17 +1100
Subject: [PATCH] add Apollo back in

Change-Id: I779fe0f245eb298b923b10a689ffa4e2edff8330

diff --git a/config/common.mk b/config/common.mk
index b2791c4..dc4d0fc 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -160,6 +160,7 @@ PRODUCT_PACKAGES += \
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
+    Apollo \
     CMFileManager \
     LockClock \
     CMAccount
