From 46e0f67dc345a313dc8f5f7f5e3cb4d2a7e5cb5d Mon Sep 17 00:00:00 2001
From: Robin Humble <plaguedbypenguins@gmail.com>
Date: Tue, 15 Oct 2013 22:22:33 +1100
Subject: [PATCH] CMWallpapers will fit, so re-add it

Change-Id: I9b4fc3e3945991625ce9302c4ca7bf0e34138369

diff --git a/config/common.mk b/config/common.mk
index 7132a45..324e859 100644
--- a/config/common.mk
+++ b/config/common.mk
@@ -166,6 +166,7 @@ PRODUCT_PACKAGES += \
     DSPManager \
     libcyanogen-dsp \
     audio_effects.conf \
+    CMWallpapers \
     Apollo \
     CMFileManager \
     LockClock \
